ui/client/components/graph/node-modal/MapVariable.tsx

```diff
@@ -1,19 +1,39 @@
 import _ from "lodash"
-import PropTypes from "prop-types"
-import {v4 as uuid4} from "uuid"
-import {errorValidator, mandatoryValueValidator} from "./editors/Validators"
+import {errorValidator, mandatoryValueValidator, Error} from "./editors/Validators"
 import LabeledInput from "./editors/field/LabeledInput"
 import LabeledTextarea from "./editors/field/LabeledTextarea"
 import Map from "./editors/map/Map"
 import React from "react"
+import {VariableTypes, NodeType, TypedObjectTypingResult, Field} from "../../../types"
 
-const MapVariable = ({isMarked, node, removeElement, addElement, onChange, readOnly, showValidation, errors, renderFieldLabel, variableTypes}) => {
+type Props = {
+  isMarked: (paths: string) => boolean,
+  node: NodeType,
+  removeElement: (namespace: string, ix: number) => void,
+  addElement: (property: $TodoType, element: $TodoType) => void,
+  onChange: (propToMutate: $TodoType, newValue: $TodoType, defaultValue?: $TodoType) => void,
+  readOnly?: boolean,
+  showValidation: boolean,
+  errors: Array<Error>,
+  variableTypes: VariableTypes,
+  renderFieldLabel: (label: string) => React.ReactNode,
+  expressionType?: TypedObjectTypingResult,
```

Comment:
do we use more narrow type somehow? Or can we have TypingResult here?

Reply:
Props to MapVariable are passed from NodeDetailsContent which is untyped yet.
I can implement type guard here, so we won't propagate invalid object (ie different than TypedObjcetTypingResult)

Reply:
ok